### PR TITLE
Show problem shares on SMS campaigns

### LINKED ISSUE #4560: add problem shares to SMS games
image

since there is only one view of SMS games (i.e. instead of pitch and action pages) the problem share should show on this one view for SMS games (whether the user is logged in or logged out)

### PR DESCRIPTION
**Pull request by sbsmith86**
Resolves #4560 . Move problem share preprocessing to `dosomething_campaign_preprocess_common_vars()` and output problem shares on the sms campaign template.

### CODE REVIEW

File: lib/modules/dosomething/dosomething_campaign/dosomething_campaign.theme.inc [hunk 1]
```diff
@@ -105,6 +105,43 @@ function dosomething_campaign_preprocess_common_vars(&$vars, &$wrapper) {
...
+    $problem_share_prompt = variable_get('dosomething_campaign_problem_share_prompt');
+    if ($problem_share_prompt) {
+      $vars['problem_share_prompt'] = $problem_share_prompt;
+    }
+
+    $base_url = url(base_path(), array('absolute'=> TRUE));
+    $campaign_path = url(current_path(), array('absolute' => TRUE));
+
+    // Create the image path to the problem fact statement image generated for the node.
+    $problem_share_image = $base_url . 'profiles/dosomething/modules/dosomething/dosomething_campaign/dosomething_campaign_problem_shares/images/' . $vars['nid'] . '.png';
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_campaign/dosomething_campaign.theme.inc]: you should [`module_get_path()`](https://www.drupal.org/node/319171) this

File: lib/modules/dosomething/dosomething_campaign/dosomething_campaign.theme.inc [hunk 2]
```diff
@@ -105,6 +105,43 @@ function dosomething_campaign_preprocess_common_vars(&$vars, &$wrapper) {
   if (isset($vars['content']['zendesk_form'])) {
     $vars['zendesk_form'] = $vars['content']['zendesk_form'];
   }
+
+  // Problem shares
+  $vars['show_problem_shares'] = theme_get_setting('show_problem_shares');
+  $campaign = $vars['campaign'];
+
+  if ($vars['show_problem_shares']) {
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_campaign/dosomething_campaign.theme.inc:113]: is this a new setting? should we set a default with this?

Comment (sbsmith86) [inline lib/modules/dosomething/dosomething_campaign/dosomething_campaign.theme.inc:113]: @angaither not new, but yes it needs a default.

File: lib/modules/dosomething/dosomething_campaign/dosomething_campaign.theme.inc (hunk 1 above)
Comment (sbsmith86) [inline lib/modules/dosomething/dosomething_campaign/dosomething_campaign.theme.inc]: @angaither My dev env is getting angry and saying `module_get_path()` is not a defined function.

Comment (sbsmith86) [inline lib/modules/dosomething/dosomething_campaign/dosomething_campaign.theme.inc]: nevermind i realized you meant `drupal_get_path`

**Comment by sbsmith86** [comment]
@angaither updated to use `drupal_get_path()`. As far as setting a default for that flag, it doesn't look like we need to since it is just a feature flag check box. It is either on or off.

**Comment by angaither** [comment]
:+1: :cool: sorry for giving you an old function!

**Pull request merged**